Uploading debug information files fails in v2.33.1

### CLI Version

2.33.1

### Operating System and Architecture

- [ ] macOS (arm64)
- [ ] macOS (x86_64)
- [ ] Linux (i686)
- [ ] Linux (x86_64)
- [ ] Linux (armv7)
- [ ] Linux (aarch64)
- [ ] Windows (i686)
- [x] Windows (x86_64)

### Operating System Version

Windows Server

### Link to reproduction repository

https://github.com/p0358/black_market_edition/actions/runs/10231364720/job/28307610591

### CLI Command

sentry-cli upload-dif --include-sources $files_to_upload

### Exact Reproduction Steps

Compile a C++ app and:
```ps
          $files_to_upload = Get-ChildItem -Path build\bin -Recurse -Include ("*.pdb", "*.dll", "*.exe")
          echo "Files to upload:" $files_to_upload
          sentry-cli upload-dif --include-sources $files_to_upload
```

### Expected Results

It should work like it did before. In v2.33.1 with PR getsentry/sentry-cli#2109 the issue getsentry/sentry-cli#2107 was reintroduced, which was previously fixed by getsentry/sentry-cli#2108, which had version v2.32.2. It looks like in both this repo and https://github.com/getsentry/symbolic/pull/816 somebody added a check for input files being UTF-8, seemingly forgetting that JavaScript is not the only language and that native apps with symbol files exist??? Barely any reasoning if at all was provided in the linked PRs as to what originally prompted such a check to be coded to begin with.

Proof for people with internal access at Sentry that it worked with symbolification and showing up source code and correct line of code: https://black-market-edition.sentry.io/issues/5651900111/?project=5545628

### Actual Results

It fails:
```
Run $files_to_upload = Get-ChildItem -Path build\bin -Recurse -Include ("*.pdb", "*.dll", "*.exe")
Files to upload:
    Directory: D:\a\black_market_edition\black_market_edition\build\bin\x64-Staging
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---            8/3/2024 11:08 PM        1876480 bme.dll
-a---            8/3/2024 11:08 PM       20779008 bme.pdb
-a---            8/3/2024 11:07 PM         389120 curl.pdb
-a---            8/3/2024 11:07 PM         946176 discord-rpc.pdb
-a---            8/3/2024 11:07 PM         921600 imgui.pdb
-a---            8/3/2024 11:07 PM         183808 launcher.dll
-a---            8/3/2024 11:07 PM        4837376 launcher.pdb
-a---            8/3/2024 11:07 PM         110592 minhook.pdb
-a---            8/3/2024 11:07 PM         176128 nghttp2.pdb
-a---            8/3/2024 11:07 PM        3534848 spdlog.pdb
-a---            8/3/2024 11:07 PM         547328 Titanfall_alt.exe
-a---            8/3/2024 11:07 PM        4419584 Titanfall_alt.pdb
> Found 6 debug information files
error: file could not be read as UTF-8
  caused by: stream did not contain valid UTF-8
Add --log-level=[info|debug] or export SENTRY_LOG_LEVEL=[info|debug] to see more output.
Please attach the full debug log to all bug reports.
Error: Process completed with exit code 1.
```

### Logs

There wouldn't be any extra  info in debug log according to what I saw in other issues

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## Unreleased
+
+- Skip invalid sources ([#861](https://github.com/getsentry/symbolic/pull/861))
+
 ## 12.10.0
 
 **Features**

--- a/symbolic-debuginfo/src/sourcebundle.rs
+++ b/symbolic-debuginfo/src/sourcebundle.rs
@@ -1314,7 +1314,12 @@ where
                 info.set_ty(SourceFileType::Source);
                 info.set_path(filename.clone());
 
-                self.add_file(bundle_path, source, info)?;
+                if let Err(e) = self.add_file(bundle_path, source, info) {
+                    // Skip sources that are not UTF-8
+                    if e.kind != SourceBundleErrorKind::ReadFailed {
+                        return Err(e);
+                    };
+                }
             }
         }
 

